backend/src/test/java/io/debezium/configserver/CreateAndDeleteMySqlConnectorIT.java

```diff
@@ -72,8 +72,7 @@ public void testMySqlDeleteConnectorSuccessful() {
                 deleteMySqlConnectorName,
                 Infrastructure.getMySqlConnectorConfiguration(1)
         );
-        Infrastructure.getDebeziumContainer().ensureConnectorTaskState(
-                deleteMySqlConnectorName, 0, Connector.State.RUNNING);
+        Infrastructure.waitForConnectorTaskStatus(deleteMySqlConnectorName, 0, Connector.State.RUNNING);
```

Comment:
So I elected to add this method to `Infrastructure`. For now this replaces the call to `ensureConnectorTaskState` because the latter introduces potential NPE issues, fixed in DBZ-5159 (as documented in the method comments), but also the ensure-method explicitly uses a timeout that isn't applicable for certain scenarios such as some PostgreSQL failures that are retried with a small delay in-between.  The tests should be resilient to these semantics and that's what this new method is meant to cover.  I adjusted several call locations in the tests to use the logic universally rather than litter the same logic in each test use case.